*** regcomp.c	2014-08-19 17:46:04.948462068 +0200
--- regcomp-pass-R-CMD-check-git2r.c	2014-08-19 17:45:39.783885395 +0200
***************
*** 18,23 ****
--- 18,27 ----
     Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
     02110-1301 USA.  */
  
+ /* Changed all 'abort' to 'Rf_error' to pass 'R CMD check git2r'*/
+ /* 2014-08-19: Stefan Widgren <stefan.widgren@gmail.com>*/
+ void Rf_error(const char*, ...);
+ 
  static reg_errcode_t re_compile_internal (regex_t *preg, const char * pattern,
  					  size_t length, reg_syntax_t syntax);
  static void re_compile_fastmap_iter (regex_t *bufp,
***************
*** 208,214 ****
  
  /* For ZOS USS we must define btowc */
  
! wchar_t 
  btowc (int c)
  {
     wchar_t wtmp[2];
--- 212,218 ----
  
  /* For ZOS USS we must define btowc */
  
! wchar_t
  btowc (int c)
  {
     wchar_t wtmp[2];
***************
*** 555,561 ****
         to this routine.  If we are given anything else, or if other regex
         code generates an invalid error code, then the program has a bug.
         Dump core so we can fix it.  */
!     abort ();
  
    msg = gettext (__re_error_msgid + __re_error_msgid_idx[errcode]);
  
--- 559,568 ----
         to this routine.  If we are given anything else, or if other regex
         code generates an invalid error code, then the program has a bug.
         Dump core so we can fix it.  */
!       Rf_error(
!           "Error in %s: Unexpected error. Please report at"
!           " https://github.com/ropensci/git2r/issues",
!           __call__);
  
    msg = gettext (__re_error_msgid + __re_error_msgid_idx[errcode]);
  
***************
*** 1094,1100 ****
  	    return;
  	break;
        default:
! 	abort ();
        }
  
    if (mb_chars || has_period)
--- 1101,1110 ----
  	    return;
  	break;
        default:
!           Rf_error(
!               "Error in %s: Unexpected error. Please report at"
!               " https://github.com/ropensci/git2r/issues",
!               __call__);
        }
  
    if (mb_chars || has_period)
