*** annotated_commit.c.orig	2016-01-08 06:56:32.024002919 +0100
--- annotated_commit.c	2016-01-08 06:57:03.484003761 +0100
***************
*** 5,10 ****
--- 5,16 ----
   * a Linking Exception. For full terms see the included COPYING file.
   */
  
+ /**
+  * Changed 'abort' to 'Rf_error' to pass 'R CMD check git2r'
+  * 2016-01-07: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ void Rf_error(const char*, ...);
+ 
  #include "common.h"
  #include "annotated_commit.h"
  #include "refs.h"
***************
*** 195,201 ****
  			git_array_clear(annotated_commit->parents);
  			break;
  		default:
! 			abort();
  	}
  
  	git__free(annotated_commit);
--- 201,209 ----
  			git_array_clear(annotated_commit->parents);
  			break;
  		default:
!                         Rf_error("Error in 'git_annotated_commit_free':"
!                                  " Unexpected error. Please report at"
!                                  " https://github.com/ropensci/git2r/issues");
  	}
  
  	git__free(annotated_commit);
