*** apply.c.orig	2016-11-19 10:03:08.118798275 +0100
--- apply.c	2016-11-19 10:03:33.138926319 +0100
***************
*** 159,164 ****
--- 159,169 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int apply_hunk(
  	patch_image *image,
  	git_patch *patch,
***************
*** 207,212 ****
--- 212,221 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int apply_hunks(
  	git_buf *out,
  	const char *source,
