*** apply.c.orig	2017-11-21 19:13:58.077751955 +0100
--- apply.c	2017-11-21 19:13:58.096751889 +0100
***************
*** 160,165 ****
--- 160,170 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int apply_hunk(
  	patch_image *image,
  	git_patch *patch,
***************
*** 208,213 ****
--- 213,222 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int apply_hunks(
  	git_buf *out,
  	const char *source,
