*** attrcache.c.orig	2017-02-27 22:07:50.154262727 +0100
--- attrcache.c	2017-02-27 22:17:53.194318999 +0100
***************
*** 305,310 ****
--- 305,315 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void attr_cache__free(git_attr_cache *cache)
  {
  	bool unlock;
***************
*** 354,359 ****
--- 359,368 ----
  	git__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__init(git_repository *repo)
  {
  	int ret = 0;
***************
*** 411,416 ****
--- 420,430 ----
  	return ret;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  void git_attr_cache_flush(git_repository *repo)
  {
  	git_attr_cache *cache;
***************
*** 422,427 ****
--- 436,445 ----
  		attr_cache__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__insert_macro(git_repository *repo, git_attr_rule *macro)
  {
  	git_attr_cache *cache = git_repository_attr_cache(repo);
***************
*** 456,459 ****
  
  	return (git_attr_rule *)git_strmap_value_at(macros, pos);
  }
- 
--- 474,476 ----
