*** attrcache.c.orig	2017-11-21 19:13:58.077751955 +0100
--- attrcache.c	2017-11-21 19:13:58.098751882 +0100
***************
*** 315,320 ****
--- 315,325 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void attr_cache__free(git_attr_cache *cache)
  {
  	bool unlock;
***************
*** 364,369 ****
--- 369,378 ----
  	git__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__init(git_repository *repo)
  {
  	int ret = 0;
***************
*** 421,426 ****
--- 430,440 ----
  	return ret;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  void git_attr_cache_flush(git_repository *repo)
  {
  	git_attr_cache *cache;
***************
*** 432,437 ****
--- 446,455 ----
  		attr_cache__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__insert_macro(git_repository *repo, git_attr_rule *macro)
  {
  	git_attr_cache *cache = git_repository_attr_cache(repo);
***************
*** 466,469 ****
  
  	return (git_attr_rule *)git_strmap_value_at(macros, pos);
  }
- 
--- 484,486 ----
