*** attrcache.c.orig	2016-11-19 16:38:12.417470911 +0100
--- attrcache.c	2016-11-19 16:39:07.525687541 +0100
***************
*** 302,307 ****
--- 302,312 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void attr_cache__free(git_attr_cache *cache)
  {
  	bool unlock;
***************
*** 351,356 ****
--- 356,365 ----
  	git__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__do_init(git_repository *repo)
  {
  	int ret = 0;
***************
*** 408,413 ****
--- 417,427 ----
  	return ret;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  void git_attr_cache_flush(git_repository *repo)
  {
  	git_attr_cache *cache;
***************
*** 419,424 ****
--- 433,442 ----
  		attr_cache__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__insert_macro(git_repository *repo, git_attr_rule *macro)
  {
  	git_attr_cache *cache = git_repository_attr_cache(repo);
***************
*** 453,456 ****
  
  	return (git_attr_rule *)git_strmap_value_at(macros, pos);
  }
- 
--- 471,473 ----
