*** attrcache.c.orig	2017-12-13 21:14:28.281920537 +0100
--- attrcache.c	2017-12-13 21:14:28.300920530 +0100
***************
*** 307,312 ****
--- 307,317 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void attr_cache__free(git_attr_cache *cache)
  {
  	bool unlock;
***************
*** 356,361 ****
--- 361,370 ----
  	git__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__init(git_repository *repo)
  {
  	int ret = 0;
***************
*** 413,418 ****
--- 422,432 ----
  	return ret;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  void git_attr_cache_flush(git_repository *repo)
  {
  	git_attr_cache *cache;
***************
*** 424,429 ****
--- 438,447 ----
  		attr_cache__free(cache);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_attr_cache__insert_macro(git_repository *repo, git_attr_rule *macro)
  {
  	git_attr_cache *cache = git_repository_attr_cache(repo);
***************
*** 458,461 ****
  
  	return (git_attr_rule *)git_strmap_value_at(macros, pos);
  }
- 
--- 476,478 ----
