*** cache.c.orig	2016-11-05 23:23:19.204028954 +0100
--- cache.c	2016-11-05 23:20:28.244024376 +0100
***************
*** 50,55 ****
--- 50,58 ----
  	if (kh_size(cache->map) == 0)
  		return;
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
  	printf("Cache %p: %d items cached, %"PRIdZ" bytes\n",
  		cache, kh_size(cache->map), cache->used_memory);
  
***************
*** 62,67 ****
--- 65,85 ----
  			object->size
  		);
  	});
+ #pragma GCC diagnostic pop
+ #else
+ 	printf("Cache %p: %d items cached, %"PRIdZ" bytes\n",
+ 		cache, kh_size(cache->map), cache->used_memory);
+ 
+ 	kh_foreach_value(cache->map, object, {
+ 		char oid_str[9];
+ 		printf(" %s%c %s (%"PRIuZ")\n",
+ 			git_object_type2string(object->type),
+ 			object->flags == GIT_CACHE_STORE_PARSED ? '*' : ' ',
+ 			git_oid_tostr(oid_str, sizeof(oid_str), &object->oid),
+ 			object->size
+ 		);
+ 	});
+ #endif
  }
  
  int git_cache_init(git_cache *cache)
