*** cache.c.orig	2016-11-19 16:42:28.299020769 +0100
--- cache.c	2016-11-19 16:42:48.579136623 +0100
***************
*** 43,48 ****
--- 43,53 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  void git_cache_dump_stats(git_cache *cache)
  {
  	git_cached_obj *object;
***************
*** 64,69 ****
--- 69,78 ----
  	});
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_cache_init(git_cache *cache)
  {
  	memset(cache, 0, sizeof(*cache));
