*** cache.c.orig	2017-02-27 22:43:49.984267970 +0100
--- cache.c	2017-02-27 22:43:50.012268330 +0100
***************
*** 41,46 ****
--- 41,51 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  void git_cache_dump_stats(git_cache *cache)
  {
  	git_cached_obj *object;
***************
*** 62,67 ****
--- 67,76 ----
  	});
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_cache_init(git_cache *cache)
  {
  	memset(cache, 0, sizeof(*cache));
