*** cc-compat.h.orig	2017-12-16 07:50:09.658695976 +0100
--- cc-compat.h	2017-12-16 07:50:27.950656805 +0100
***************
*** 46,52 ****
  #define GIT_UNUSED(x) ((void)(x))
  
  /* Define the printf format specifer to use for size_t output */
! #if defined(_MSC_VER) || defined(__MINGW32__)
  #	define PRIuZ "Iu"
  #	define PRIxZ "Ix"
  #	define PRIdZ "Id"
--- 46,52 ----
  #define GIT_UNUSED(x) ((void)(x))
  
  /* Define the printf format specifer to use for size_t output */
! #if defined(_MSC_VER)
  #	define PRIuZ "Iu"
  #	define PRIxZ "Ix"
  #	define PRIdZ "Id"
