*** checkout.c.orig	2016-11-19 10:10:40.373132727 +0100
--- checkout.c	2016-11-19 10:11:55.697525029 +0100
***************
*** 1269,1274 ****
--- 1269,1279 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int checkout_get_actions(
  	uint32_t **actions_ptr,
  	size_t **counts_ptr,
***************
*** 1364,1369 ****
--- 1369,1378 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static bool should_remove_existing(checkout_data *data)
  {
  	int ignorecase;
