*** checkout.c.orig	2016-01-07 22:35:56.012078704 +0100
--- checkout.c	2016-01-07 22:35:56.156078708 +0100
***************
*** 5,10 ****
--- 5,16 ----
   * a Linking Exception. For full terms see the included COPYING file.
   */
  
+ /**
+  * Changed 'abort' to 'Rf_error' to pass 'R CMD check git2r'
+  * 2014-11-20: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ void Rf_error(const char*, ...);
+ 
  #include <assert.h>
  
  #include "checkout.h"
***************
*** 1203,1209 ****
  	else if (theirs)
  		name = git__strdup(theirs->path);
  	else
! 		abort();
  
  	GITERR_CHECK_ALLOC(name);
  
--- 1209,1217 ----
  	else if (theirs)
  		name = git__strdup(theirs->path);
  	else
!                 Rf_error("Unexpected error in 'checkout_conflict_append_remove'."
!                          " Please report at"
!                          " https://github.com/ropensci/git2r/issues");
  
  	GITERR_CHECK_ALLOC(name);
  
