*** checkout.c.orig	2017-07-09 11:32:12.238269212 +0200
--- checkout.c	2017-07-09 11:32:12.274268997 +0200
***************
*** 1265,1270 ****
--- 1265,1275 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int checkout_get_actions(
  	uint32_t **actions_ptr,
  	size_t **counts_ptr,
***************
*** 1360,1365 ****
--- 1365,1374 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static bool should_remove_existing(checkout_data *data)
  {
  	int ignorecase;
