*** checkout.c.orig	2017-03-02 20:16:01.260304175 +0100
--- checkout.c	2017-03-02 20:16:01.324305149 +0100
***************
*** 1267,1272 ****
--- 1267,1277 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int checkout_get_actions(
  	uint32_t **actions_ptr,
  	size_t **counts_ptr,
***************
*** 1362,1367 ****
--- 1367,1376 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static bool should_remove_existing(checkout_data *data)
  {
  	int ignorecase;
