*** common.h.orig	2017-12-14 21:45:02.475918727 +0100
--- common.h	2017-12-14 21:45:29.771752750 +0100
***************
*** 40,46 ****
  # include <winsock2.h>
  # include <windows.h>
  # include <ws2tcpip.h>
! # include "win32/msvc-compat.h"
  # include "win32/mingw-compat.h"
  # include "win32/win32-compat.h"
  # include "win32/error.h"
--- 40,46 ----
  # include <winsock2.h>
  # include <windows.h>
  # include <ws2tcpip.h>
! /* # include "win32/msvc-compat.h" */
  # include "win32/mingw-compat.h"
  # include "win32/win32-compat.h"
  # include "win32/error.h"
