*** common.h.orig	2016-01-08 07:35:38.696003045 +0100
--- common.h	2016-01-08 07:37:15.428005635 +0100
***************
*** 7,12 ****
--- 7,20 ----
  #ifndef INCLUDE_common_h__
  #define INCLUDE_common_h__
  
+ /**
+  * Removed unused include directives
+  *  - # include "win32/msvc-compat.h"
+  *  - # include "win32/mingw-compat.h"
+  *
+  * 2014-12-26: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ 
  #include "git2/common.h"
  #include "cc-compat.h"
  
***************
*** 39,46 ****
  # include <winsock2.h>
  # include <windows.h>
  # include <ws2tcpip.h>
- # include "win32/msvc-compat.h"
- # include "win32/mingw-compat.h"
  # include "win32/win32-compat.h"
  # include "win32/error.h"
  # include "win32/version.h"
--- 47,52 ----
