*** common.h	2015-03-16 17:57:21.820437872 +0100
--- common-remove-includes.h	2015-03-16 17:57:26.108437988 +0100
***************
*** 7,12 ****
--- 7,20 ----
  #ifndef INCLUDE_common_h__
  #define INCLUDE_common_h__
  
+ /**
+  * Removed unused include directives
+  *  - # include "win32/msvc-compat.h"
+  *  - # include "win32/mingw-compat.h"
+  *
+  * 2014-12-26: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ 
  #include "git2/common.h"
  #include "cc-compat.h"
  
***************
*** 39,46 ****
  # include <winsock2.h>
  # include <windows.h>
  # include <ws2tcpip.h>
- # include "win32/msvc-compat.h"
- # include "win32/mingw-compat.h"
  # include "win32/error.h"
  # include "win32/version.h"
  # ifdef GIT_THREADS
--- 47,52 ----
