*** config.c.orig	2017-07-09 11:32:12.246269164 +0200
--- config.c	2017-07-09 11:32:12.278268973 +0200
***************
*** 627,633 ****
--- 627,640 ----
  int git_config_set_int64(git_config *cfg, const char *name, int64_t value)
  {
  	char str_value[32]; /* All numbers should fit in here */
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat-extra-args"
  	p_snprintf(str_value, sizeof(str_value), "%" PRId64, value);
+ #pragma GCC diagnostic pop
+ #else
+ 	p_snprintf(str_value, sizeof(str_value), "%" PRId64, value);
+ #endif
  	return git_config_set_string(cfg, name, str_value);
  }
  
