*** diff.c.orig	2017-07-09 11:32:12.238269212 +0200
--- diff.c	2017-07-09 11:32:12.282268949 +0200
***************
*** 155,160 ****
--- 155,165 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_diff_format_email__append_header_tobuf(
  	git_buf *out,
  	const git_oid *id,
***************
*** 212,217 ****
--- 217,226 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_diff_format_email__append_patches_tobuf(
  	git_buf *out,
  	git_diff *diff)
***************
*** 236,241 ****
--- 245,255 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_diff_format_email(
  	git_buf *out,
  	git_diff *diff,
***************
*** 317,322 ****
--- 331,340 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_diff_commit_as_email(
  	git_buf *out,
  	git_repository *repo,
***************
*** 373,376 ****
  		GIT_DIFF_FORMAT_EMAIL_OPTIONS_INIT);
  	return 0;
  }
- 
--- 391,393 ----
