*** diff_stats.c.orig	2016-11-19 20:41:55.304405864 +0100
--- diff_stats.c	2016-11-19 20:44:35.417572927 +0100
***************
*** 44,49 ****
--- 44,54 ----
  	return count;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_diff_file_stats__full_to_buf(
  	git_buf *out,
  	const git_diff_delta *delta,
***************
*** 134,139 ****
--- 139,148 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_diff_file_stats__summary_to_buf(
  	git_buf *out,
  	const git_diff_delta *delta)
***************
*** 254,259 ****
--- 263,273 ----
  	return stats->deletions;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_diff_stats_to_buf(
  	git_buf *out,
  	const git_diff_stats *stats,
***************
*** 334,339 ****
--- 348,357 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  void git_diff_stats_free(git_diff_stats *stats)
  {
  	if (stats == NULL)
