*** diff.h.orig	2018-04-25 18:41:08.883493589 +0200
--- diff.h	2018-04-25 18:41:29.219426486 +0200
***************
*** 211,217 ****
  	 * which generally can produce better diffs when dealing with ambiguous
  	 * diff hunks.
  	 */
! 	GIT_DIFF_INDENT_HEURISTIC = (1u << 31),
  } git_diff_option_t;
  
  /**
--- 211,217 ----
  	 * which generally can produce better diffs when dealing with ambiguous
  	 * diff hunks.
  	 */
! 	GIT_DIFF_INDENT_HEURISTIC = (1 << 31),
  } git_diff_option_t;
  
  /**
