*** diff_stats.c.orig	2016-11-05 23:23:19.196028954 +0100
--- diff_stats.c	2016-11-05 23:20:51.628025002 +0100
***************
*** 79,93 ****
--- 79,112 ----
  		goto on_error;
  
  	if (delta->flags & GIT_DIFF_FLAG_BINARY) {
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
  		if (git_buf_printf(out,
  				"Bin %" PRIuZ " -> %" PRIuZ " bytes", old_size, new_size) < 0)
  			goto on_error;
+ #pragma GCC diagnostic pop
+ #else
+ 		if (git_buf_printf(out,
+ 				"Bin %" PRIuZ " -> %" PRIuZ " bytes", old_size, new_size) < 0)
+ 			goto on_error;
+ #endif
  	}
  	else {
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ 		if (git_buf_printf(out,
+ 				"%*" PRIuZ, stats->max_digits,
+ 				filestat->insertions + filestat->deletions) < 0)
+ 			goto on_error;
+ #pragma GCC diagnostic pop
+ #else
  		if (git_buf_printf(out,
  				"%*" PRIuZ, stats->max_digits,
  				filestat->insertions + filestat->deletions) < 0)
  			goto on_error;
+ #endif
  
  		if (filestat->insertions || filestat->deletions) {
  			if (git_buf_putc(out, ' ') < 0)
***************
*** 125,135 ****
  	int error;
  	const char *path = delta->new_file.path;
  
! 	if (delta->flags & GIT_DIFF_FLAG_BINARY)
  		error = git_buf_printf(out, "%-8c" "%-8c" "%s\n", '-', '-', path);
! 	else
  		error = git_buf_printf(out, "%-8" PRIuZ "%-8" PRIuZ "%s\n",
  			filestats->insertions, filestats->deletions, path);
  
  	return error;
  }
--- 144,163 ----
  	int error;
  	const char *path = delta->new_file.path;
  
! 	if (delta->flags & GIT_DIFF_FLAG_BINARY) {
  		error = git_buf_printf(out, "%-8c" "%-8c" "%s\n", '-', '-', path);
! 	} else {
! #ifdef _WIN32
! #pragma GCC diagnostic push
! #pragma GCC diagnostic ignored "-Wformat"
! 		error = git_buf_printf(out, "%-8" PRIuZ "%-8" PRIuZ "%s\n",
! 			filestats->insertions, filestats->deletions, path);
! #pragma GCC diagnostic pop
! #else
  		error = git_buf_printf(out, "%-8" PRIuZ "%-8" PRIuZ "%s\n",
  			filestats->insertions, filestats->deletions, path);
+ #endif
+         }
  
  	return error;
  }
***************
*** 300,305 ****
--- 328,351 ----
  	}
  
  	if (format & GIT_DIFF_STATS_FULL || format & GIT_DIFF_STATS_SHORT) {
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ 		git_buf_printf(
+ 			out, " %" PRIuZ " file%s changed",
+ 			stats->files_changed, stats->files_changed != 1 ? "s" : "");
+ 
+ 		if (stats->insertions || stats->deletions == 0)
+ 			git_buf_printf(
+ 				out, ", %" PRIuZ " insertion%s(+)",
+ 				stats->insertions, stats->insertions != 1 ? "s" : "");
+ 
+ 		if (stats->deletions || stats->insertions == 0)
+ 			git_buf_printf(
+ 				out, ", %" PRIuZ " deletion%s(-)",
+ 				stats->deletions, stats->deletions != 1 ? "s" : "");
+ #pragma GCC diagnostic pop
+ #else
  		git_buf_printf(
  			out, " %" PRIuZ " file%s changed",
  			stats->files_changed, stats->files_changed != 1 ? "s" : "");
***************
*** 313,318 ****
--- 359,365 ----
  			git_buf_printf(
  				out, ", %" PRIuZ " deletion%s(-)",
  				stats->deletions, stats->deletions != 1 ? "s" : "");
+ #endif
  
  		git_buf_putc(out, '\n');
  
