*** diff.c.orig	2017-11-21 19:13:58.078751951 +0100
--- diff.c	2017-11-21 19:13:58.110751841 +0100
***************
*** 156,161 ****
--- 156,166 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_diff_format_email__append_header_tobuf(
  	git_buf *out,
  	const git_oid *id,
***************
*** 213,218 ****
--- 218,227 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_diff_format_email__append_patches_tobuf(
  	git_buf *out,
  	git_diff *diff)
***************
*** 237,242 ****
--- 246,256 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_diff_format_email(
  	git_buf *out,
  	git_diff *diff,
***************
*** 318,323 ****
--- 332,341 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_diff_commit_as_email(
  	git_buf *out,
  	git_repository *repo,
***************
*** 374,377 ****
  		GIT_DIFF_FORMAT_EMAIL_OPTIONS_INIT);
  	return 0;
  }
- 
--- 392,394 ----
