*** diff_print.c.orig	2017-11-21 19:13:58.079751948 +0100
--- diff_print.c	2017-11-21 19:13:58.112751834 +0100
***************
*** 451,456 ****
--- 451,461 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int format_binary(
  	diff_print_info *pi,
  	git_diff_binary_t type,
***************
*** 489,494 ****
--- 494,503 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int diff_print_patch_file_binary_noshow(
  	diff_print_info *pi, git_diff_delta *delta,
  	const char *old_pfx, const char *new_pfx)
