*** fetchhead.c.orig	2016-11-19 10:24:28.693483205 +0100
--- fetchhead.c	2016-11-19 10:25:55.129940718 +0100
***************
*** 133,138 ****
--- 133,143 ----
  	return git_filebuf_commit(&file);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int fetchhead_ref_parse(
  	git_oid *oid,
  	unsigned int *is_merge,
***************
*** 290,295 ****
--- 295,304 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  void git_fetchhead_ref_free(git_fetchhead_ref *fetchhead_ref)
  {
  	if (fetchhead_ref == NULL)
