*** global.c.orig	2016-11-19 20:51:50.140023859 +0100
--- global.c	2016-11-19 20:53:51.172656957 +0100
***************
*** 70,75 ****
--- 70,80 ----
  	return ret;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void shutdown_common(void)
  {
  	int pos;
***************
*** 90,95 ****
--- 95,104 ----
  	git__free(git__ssl_ciphers);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  /**
   * Handle the global state with TLS
   *
