*** global.c.orig	2017-11-21 19:13:58.079751948 +0100
--- global.c	2017-11-21 19:13:58.117751816 +0100
***************
*** 71,76 ****
--- 71,81 ----
  	return ret;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void shutdown_common(void)
  {
  	int pos;
***************
*** 91,96 ****
--- 96,105 ----
  	git__free(git__ssl_ciphers);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  /**
   * Handle the global state with TLS
   *
