*** index.c.orig	2017-07-09 11:34:35.565411307 +0200
--- index.c	2017-07-09 11:34:35.601411090 +0200
***************
*** 480,485 ****
--- 480,490 ----
  	GIT_REFCOUNT_DEC(index, index_free);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  /* call with locked index */
  static void index_free_deleted(git_index *index)
  {
***************
*** 497,502 ****
--- 502,511 ----
  	git_vector_clear(&index->deleted);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  /* call with locked index */
  static int index_remove_entry(git_index *index, size_t pos)
  {
***************
*** 2132,2137 ****
--- 2141,2151 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  void git_index_reuc_clear(git_index *index)
  {
  	size_t i;
***************
*** 2144,2149 ****
--- 2158,2167 ----
  	git_vector_clear(&index->reuc);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int index_error_invalid(const char *message)
  {
  	giterr_set(GITERR_INDEX, "invalid data in index - %s", message);
***************
*** 2981,2986 ****
--- 2999,3009 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  int git_index_read_tree(git_index *index, const git_tree *tree)
  {
  	int error = 0;
***************
*** 3182,3187 ****
--- 3205,3214 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_index_read_index(
  	git_index *index,
  	const git_index *new_index)
