*** index.c.orig	2016-11-05 23:23:19.184028953 +0100
--- index.c	2016-11-05 23:21:06.152025391 +0100
***************
*** 497,503 ****
--- 497,510 ----
  		return;
  
  	for (i = 0; i < index->deleted.length; ++i) {
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
  		git_index_entry *ie = git__swap(index->deleted.contents[i], NULL);
+ #pragma GCC diagnostic pop
+ #else
+ 		git_index_entry *ie = git__swap(index->deleted.contents[i], NULL);
+ #endif
  		index_entry_free(ie);
  	}
  
***************
*** 2145,2152 ****
  
  	assert(index);
  
! 	for (i = 0; i < index->reuc.length; ++i)
  		index_entry_reuc_free(git__swap(index->reuc.contents[i], NULL));
  
  	git_vector_clear(&index->reuc);
  }
--- 2152,2167 ----
  
  	assert(index);
  
! 	for (i = 0; i < index->reuc.length; ++i) {
! #ifdef _WIN32
! #pragma GCC diagnostic push
! #pragma GCC diagnostic ignored "-Wpedantic"
! 		index_entry_reuc_free(git__swap(index->reuc.contents[i], NULL));
! #pragma GCC diagnostic pop
! #else
  		index_entry_reuc_free(git__swap(index->reuc.contents[i], NULL));
+ #endif
+         }
  
  	git_vector_clear(&index->reuc);
  }
***************
*** 3000,3006 ****
--- 3015,3028 ----
  		/* well, this isn't good */;
  	} else {
  		git_vector_swap(&entries, &index->entries);
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
  		entries_map = git__swap(index->entries_map, entries_map);
+ #pragma GCC diagnostic pop
+ #else
+ 		entries_map = git__swap(index->entries_map, entries_map);
+ #endif
  	}
  
  cleanup:
***************
*** 3131,3137 ****
--- 3153,3166 ----
  	git_index_reuc_clear(index);
  
  	git_vector_swap(&new_entries, &index->entries);
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
  	new_entries_map = git__swap(index->entries_map, new_entries_map);
+ #pragma GCC diagnostic pop
+ #else
+ 	new_entries_map = git__swap(index->entries_map, new_entries_map);
+ #endif
  
  	git_vector_foreach(&remove_entries, i, entry) {
  		if (index->tree)
