*** index.c.orig	2017-03-02 20:16:01.280304479 +0100
--- index.c	2017-03-02 20:16:01.340305393 +0100
***************
*** 484,489 ****
--- 484,494 ----
  	GIT_REFCOUNT_DEC(index, index_free);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  /* call with locked index */
  static void index_free_deleted(git_index *index)
  {
***************
*** 501,506 ****
--- 506,515 ----
  	git_vector_clear(&index->deleted);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  /* call with locked index */
  static int index_remove_entry(git_index *index, size_t pos)
  {
***************
*** 2136,2141 ****
--- 2145,2155 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  void git_index_reuc_clear(git_index *index)
  {
  	size_t i;
***************
*** 2148,2153 ****
--- 2162,2171 ----
  	git_vector_clear(&index->reuc);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int index_error_invalid(const char *message)
  {
  	giterr_set(GITERR_INDEX, "invalid data in index - %s", message);
***************
*** 2948,2953 ****
--- 2966,2976 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  int git_index_read_tree(git_index *index, const git_tree *tree)
  {
  	int error = 0;
***************
*** 3149,3154 ****
--- 3172,3181 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_index_read_index(
  	git_index *index,
  	const git_index *new_index)
