*** iterator.c.orig	2017-11-21 19:13:58.080751944 +0100
--- iterator.c	2017-11-21 19:13:58.121751803 +0100
***************
*** 1297,1302 ****
--- 1297,1307 ----
  	return entry;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int filesystem_iterator_frame_push(
  	filesystem_iterator *iter,
  	filesystem_iterator_entry *frame_entry)
***************
*** 1439,1444 ****
--- 1444,1453 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  GIT_INLINE(void) filesystem_iterator_frame_pop(filesystem_iterator *iter)
  {
  	filesystem_iterator_frame *frame;
