*** merge.c.orig	2017-07-09 11:34:35.553411379 +0200
--- merge.c	2017-07-09 11:34:35.605411066 +0200
***************
*** 52,57 ****
--- 52,61 ----
  #define GIT_MERGE_INDEX_ENTRY_EXISTS(X)	((X).mode != 0)
  #define GIT_MERGE_INDEX_ENTRY_ISFILE(X) S_ISREG((X).mode)
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
  
  typedef enum {
  	TREE_IDX_ANCESTOR = 0,
***************
*** 3192,3194 ****
--- 3196,3202 ----
  		opts, version, git_merge_file_options, GIT_MERGE_FILE_OPTIONS_INIT);
  	return 0;
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
