*** odb.c.orig	2017-07-09 11:34:35.553411379 +0200
--- odb.c	2017-07-09 11:36:59.648544838 +0200
***************
*** 77,82 ****
--- 77,87 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat-extra-args"
+ #endif
+ 
  int git_odb__format_object_header(char *hdr, size_t n, git_off_t obj_len, git_otype obj_type)
  {
  	const char *type_str = git_object_type2string(obj_type);
***************
*** 85,90 ****
--- 90,99 ----
  	return len+1;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_odb__hashobj(git_oid *id, git_rawobj *obj)
  {
  	git_buf_vec vec[2];
***************
*** 481,486 ****
--- 490,499 ----
  	return -1;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
  
  int git_odb_get_backend(git_odb_backend **out, git_odb *odb, size_t pos)
  {
***************
*** 498,503 ****
--- 511,520 ----
  	return GIT_ENOTFOUND;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_odb__add_default_backends(
  	git_odb *db, const char *objects_dir,
  	bool as_alternates, int alternate_depth)
***************
*** 1323,1328 ****
--- 1340,1350 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int git_odb_stream__invalid_length(
  	const git_odb_stream *stream,
  	const char *action)
***************
*** 1336,1341 ****
--- 1358,1367 ----
  	return -1;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_odb_stream_write(git_odb_stream *stream, const char *buffer, size_t len)
  {
  	git_hash_update(stream->hash_ctx, buffer, len);
