*** patch_parse.c.orig	2017-07-09 11:38:26.948018311 +0200
--- patch_parse.c	2017-07-09 11:38:27.000017998 +0200
***************
*** 10,15 ****
--- 10,20 ----
  #include "diff_parse.h"
  #include "path.h"
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  #define parse_err(...) \
  	( giterr_set(GITERR_PATCH, __VA_ARGS__), -1 )
  
***************
*** 1158,1160 ****
--- 1163,1168 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
