*** patch_parse.c.orig	2017-11-21 19:13:58.081751941 +0100
--- patch_parse.c	2017-11-21 19:13:58.127751782 +0100
***************
*** 12,17 ****
--- 12,22 ----
  #include "diff_parse.h"
  #include "path.h"
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  #define parse_err(...) \
  	( giterr_set(GITERR_PATCH, __VA_ARGS__), -1 )
  
***************
*** 1160,1162 ****
--- 1165,1170 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
