*** posix.h.orig	2017-12-14 19:08:50.712214873 +0100
--- posix.h	2017-12-14 19:09:12.176164382 +0100
***************
*** 7,13 ****
  #ifndef INCLUDE_posix__unix_h__
  #define INCLUDE_posix__unix_h__
  
! #include "git2/sys/features.h"
  #include <stdio.h>
  #include <dirent.h>
  #include <sys/param.h>
--- 7,13 ----
  #ifndef INCLUDE_posix__unix_h__
  #define INCLUDE_posix__unix_h__
  
! /* #include "git2/sys/features.h" */
  #include <stdio.h>
  #include <dirent.h>
  #include <sys/param.h>
