*** posix.h	2015-06-28 11:34:36.901255415 +0200
--- posix-pass-R-CMD-check-git2r.h	2015-06-28 11:34:36.953255416 +0200
***************
*** 7,12 ****
--- 7,19 ----
  #ifndef INCLUDE_posix__w32_h__
  #define INCLUDE_posix__w32_h__
  
+ /**
+  * Changed to use the 'gnu_printf' format attribute on 'p_snprintf' to
+  * pass 'R CMD check git2r' on Windows
+  *
+  * 2014-08-22: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ 
  #include "common.h"
  #include "../posix.h"
  #include "path_w32.h"
***************
*** 36,42 ****
  extern int p_inet_pton(int af, const char* src, void* dst);
  
  extern int p_vsnprintf(char *buffer, size_t count, const char *format, va_list argptr);
! extern int p_snprintf(char *buffer, size_t count, const char *format, ...) GIT_FORMAT_PRINTF(3, 4);
  extern int p_mkstemp(char *tmp_path);
  extern int p_chdir(const char* path);
  extern int p_chmod(const char* path, mode_t mode);
--- 43,49 ----
  extern int p_inet_pton(int af, const char* src, void* dst);
  
  extern int p_vsnprintf(char *buffer, size_t count, const char *format, va_list argptr);
! extern int p_snprintf(char *buffer, size_t count, const char *format, ...) __attribute__((__format__ (gnu_printf, 3, 4)));
  extern int p_mkstemp(char *tmp_path);
  extern int p_chdir(const char* path);
  extern int p_chmod(const char* path, mode_t mode);
