*** rebase.c	2015-05-14 13:14:04.774170802 +0200
--- rebase-pass-R-CMD-check-git2r.c	2015-05-14 13:13:47.266170655 +0200
***************
*** 5,10 ****
--- 5,16 ----
   * a Linking Exception. For full terms see the included COPYING file.
   */
  
+ /**
+  * Changed all 'abort' to 'Rf_error' to pass 'R CMD check git2r'
+  * 2014-11-18: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ void Rf_error(const char*, ...);
+ 
  #include "common.h"
  #include "buffer.h"
  #include "repository.h"
***************
*** 375,381 ****
  		error = -1;
  		break;
  	default:
! 		abort();
  	}
  
  done:
--- 381,389 ----
  		error = -1;
  		break;
  	default:
!             Rf_error(
!                 "Error in 'git_rebase_open': Unexpected error. Please report at"
!                 " https://github.com/ropensci/git2r/issues");
  	}
  
  done:
***************
*** 730,736 ****
  		if (!checkout_opts->their_label)
  			checkout_opts->their_label = git_commit_summary(current_commit);
  	} else {
! 		abort();
  	}
  }
  
--- 738,746 ----
  		if (!checkout_opts->their_label)
  			checkout_opts->their_label = git_commit_summary(current_commit);
  	} else {
!             Rf_error(
!                 "Error in 'normalize_checkout_opts': Unexpected error. Please report at"
!                 " https://github.com/ropensci/git2r/issues");
  	}
  }
  
***************
*** 825,831 ****
  		error = rebase_next_merge(out, rebase);
  		break;
  	default:
! 		abort();
  	}
  
  	return error;
--- 835,843 ----
  		error = rebase_next_merge(out, rebase);
  		break;
  	default:
!             Rf_error(
!                 "Error in 'git_rebase_next': Unexpected error. Please report at"
!                 " https://github.com/ropensci/git2r/issues");
  	}
  
  	return error;
***************
*** 934,940 ****
  			id, rebase, author, committer, message_encoding, message);
  		break;
  	default:
! 		abort();
  	}
  
  	return error;
--- 946,954 ----
  			id, rebase, author, committer, message_encoding, message);
  		break;
  	default:
!             Rf_error(
!                 "Error in 'git_rebase_commit': Unexpected error. Please report at"
!                 " https://github.com/ropensci/git2r/issues");
  	}
  
  	return error;
