*** rebase.c.orig	2016-11-19 21:13:48.354687538 +0100
--- rebase.c	2016-11-19 21:16:37.179479268 +0100
***************
*** 198,203 ****
--- 198,208 ----
  	return operation;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int rebase_open_merge(git_rebase *rebase)
  {
  	git_buf state_path = GIT_BUF_INIT, buf = GIT_BUF_INIT, cmt = GIT_BUF_INIT;
***************
*** 257,262 ****
--- 262,271 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int rebase_alloc(git_rebase **out, const git_rebase_options *rebase_opts)
  {
  	git_rebase *rebase = git__calloc(1, sizeof(git_rebase));
***************
*** 439,444 ****
--- 448,458 ----
  		return onto->id_str;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int rebase_setupfiles_merge(git_rebase *rebase)
  {
  	git_buf commit_filename = GIT_BUF_INIT;
***************
*** 469,474 ****
--- 483,492 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int rebase_setupfiles(git_rebase *rebase)
  {
  	char onto[GIT_OID_HEXSZ], orig_head[GIT_OID_HEXSZ];
***************
*** 782,787 ****
--- 800,810 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int rebase_next_merge(
  	git_rebase_operation **out,
  	git_rebase *rebase)
***************
*** 844,849 ****
--- 867,876 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int rebase_next_inmemory(
  	git_rebase_operation **out,
  	git_rebase *rebase)
