*** reflog.c.orig	2017-11-21 19:13:58.082751937 +0100
--- reflog.c	2017-11-21 19:13:58.131751768 +0100
***************
*** 184,189 ****
--- 184,194 ----
  	return entry->msg;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_reflog_drop(git_reflog *reflog, size_t idx, int rewrite_previous_entry)
  {
  	size_t entrycount;
***************
*** 231,233 ****
--- 236,242 ----
  
  	return 0;
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
