*** reparse.h.orig	2017-12-17 15:40:08.887769158 +0100
--- reparse.h	2017-12-17 15:40:26.360711777 +0100
***************
*** 18,24 ****
  	ULONG  ReparseTag;
  	USHORT ReparseDataLength;
  	USHORT Reserved;
! 	union {
  		struct {
  			USHORT SubstituteNameOffset;
  			USHORT SubstituteNameLength;
--- 18,24 ----
  	ULONG  ReparseTag;
  	USHORT ReparseDataLength;
  	USHORT Reserved;
! 	_ANONYMOUS_UNION union {
  		struct {
  			USHORT SubstituteNameOffset;
  			USHORT SubstituteNameLength;
***************
*** 37,43 ****
  		struct {
  			UCHAR DataBuffer[1];
  		} GenericReparseBuffer;
! 	};
  } GIT_REPARSE_DATA_BUFFER;
  
  #define REPARSE_DATA_HEADER_SIZE			8
--- 37,43 ----
  		struct {
  			UCHAR DataBuffer[1];
  		} GenericReparseBuffer;
! 	} DUMMYUNIONNAME;
  } GIT_REPARSE_DATA_BUFFER;
  
  #define REPARSE_DATA_HEADER_SIZE			8
