*** reparse.h.orig	2016-11-05 23:23:19.168028953 +0100
--- reparse.h	2016-11-05 23:21:59.364026816 +0100
***************
*** 14,19 ****
--- 14,22 ----
  * It was formerly included in the Windows 2000 SDK and remains defined in
  * MinGW, so we must define it with a silly name to avoid conflicting.
  */
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
  typedef struct _GIT_REPARSE_DATA_BUFFER {
  	ULONG  ReparseTag;
  	USHORT ReparseDataLength;
***************
*** 39,44 ****
--- 42,49 ----
  		} GenericReparseBuffer;
  	};
  } GIT_REPARSE_DATA_BUFFER;
+ #pragma GCC diagnostic pop
+ #endif
  
  #define REPARSE_DATA_HEADER_SIZE			8
  #define REPARSE_DATA_MOUNTPOINT_HEADER_SIZE	8
