*** repository.c.orig	2017-11-21 19:13:58.077751955 +0100
--- repository.c	2017-11-21 19:13:58.133751761 +0100
***************
*** 83,88 ****
--- 83,93 ----
  };
  size_t git_repository__reserved_names_posix_len = 1;
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void set_odb(git_repository *repo, git_odb *odb)
  {
  	if (odb) {
***************
*** 137,142 ****
--- 142,151 ----
  	}
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  void git_repository__cleanup(git_repository *repo)
  {
  	assert(repo);
***************
*** 2836,2841 ****
--- 2845,2855 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  int git_repository_set_ident(git_repository *repo, const char *name, const char *email)
  {
  	char *tmp_name = NULL, *tmp_email = NULL;
***************
*** 2886,2888 ****
--- 2900,2906 ----
  	repo->submodule_cache = 0;
  	return 0;
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
