*** repository.c.orig	2016-11-05 23:23:19.180028953 +0100
--- repository.c	2016-11-05 23:21:19.276025742 +0100
***************
*** 58,67 ****
--- 58,77 ----
  		GIT_REFCOUNT_INC(odb);
  	}
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
  	if ((odb = git__swap(repo->_odb, odb)) != NULL) {
  		GIT_REFCOUNT_OWN(odb, NULL);
  		git_odb_free(odb);
  	}
+ #pragma GCC diagnostic pop
+ #else
+ 	if ((odb = git__swap(repo->_odb, odb)) != NULL) {
+ 		GIT_REFCOUNT_OWN(odb, NULL);
+ 		git_odb_free(odb);
+ 	}
+ #endif
  }
  
  static void set_refdb(git_repository *repo, git_refdb *refdb)
***************
*** 71,80 ****
--- 81,100 ----
  		GIT_REFCOUNT_INC(refdb);
  	}
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ 	if ((refdb = git__swap(repo->_refdb, refdb)) != NULL) {
+ 		GIT_REFCOUNT_OWN(refdb, NULL);
+ 		git_refdb_free(refdb);
+ 	}
+ #pragma GCC diagnostic pop
+ #else
  	if ((refdb = git__swap(repo->_refdb, refdb)) != NULL) {
  		GIT_REFCOUNT_OWN(refdb, NULL);
  		git_refdb_free(refdb);
  	}
+ #endif
  }
  
  static void set_config(git_repository *repo, git_config *config)
***************
*** 84,93 ****
--- 104,123 ----
  		GIT_REFCOUNT_INC(config);
  	}
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ 	if ((config = git__swap(repo->_config, config)) != NULL) {
+ 		GIT_REFCOUNT_OWN(config, NULL);
+ 		git_config_free(config);
+ 	}
+ #pragma GCC diagnostic pop
+ #else
  	if ((config = git__swap(repo->_config, config)) != NULL) {
  		GIT_REFCOUNT_OWN(config, NULL);
  		git_config_free(config);
  	}
+ #endif
  
  	git_repository__cvar_cache_clear(repo);
  }
***************
*** 99,108 ****
--- 129,148 ----
  		GIT_REFCOUNT_INC(index);
  	}
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ 	if ((index = git__swap(repo->_index, index)) != NULL) {
+ 		GIT_REFCOUNT_OWN(index, NULL);
+ 		git_index_free(index);
+ 	}
+ #pragma GCC diagnostic pop
+ #else
  	if ((index = git__swap(repo->_index, index)) != NULL) {
  		GIT_REFCOUNT_OWN(index, NULL);
  		git_index_free(index);
  	}
+ #endif
  }
  
  void git_repository__cleanup(git_repository *repo)
***************
*** 2530,2537 ****
--- 2570,2585 ----
  		GITERR_CHECK_ALLOC(tmp_email);
  	}
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
  	tmp_name = git__swap(repo->ident_name, tmp_name);
  	tmp_email = git__swap(repo->ident_email, tmp_email);
+ #pragma GCC diagnostic pop
+ #else
+ 	tmp_name = git__swap(repo->ident_name, tmp_name);
+ 	tmp_email = git__swap(repo->ident_email, tmp_email);
+ #endif
  
  	git__free(tmp_name);
  	git__free(tmp_email);
