*** repository.c.orig	2016-11-19 21:22:55.533276035 +0100
--- repository.c	2016-11-19 21:26:30.474323932 +0100
***************
*** 51,56 ****
--- 51,61 ----
  };
  size_t git_repository__reserved_names_posix_len = 1;
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void set_odb(git_repository *repo, git_odb *odb)
  {
  	if (odb) {
***************
*** 105,110 ****
--- 110,119 ----
  	}
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  void git_repository__cleanup(git_repository *repo)
  {
  	assert(repo);
***************
*** 2519,2524 ****
--- 2528,2538 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  int git_repository_set_ident(git_repository *repo, const char *name, const char *email)
  {
  	char *tmp_name = NULL, *tmp_email = NULL;
***************
*** 2541,2543 ****
--- 2555,2561 ----
  
  	return 0;
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
