*** repository.c.orig	2017-07-09 11:38:26.948018311 +0200
--- repository.c	2017-07-09 11:38:27.004017974 +0200
***************
*** 81,86 ****
--- 81,91 ----
  };
  size_t git_repository__reserved_names_posix_len = 1;
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void set_odb(git_repository *repo, git_odb *odb)
  {
  	if (odb) {
***************
*** 135,140 ****
--- 140,149 ----
  	}
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  void git_repository__cleanup(git_repository *repo)
  {
  	assert(repo);
***************
*** 2828,2833 ****
--- 2837,2847 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  int git_repository_set_ident(git_repository *repo, const char *name, const char *email)
  {
  	char *tmp_name = NULL, *tmp_email = NULL;
***************
*** 2878,2880 ****
--- 2892,2898 ----
  	repo->submodule_cache = 0;
  	return 0;
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
