*** repository.c.orig	2017-03-02 20:16:01.292304662 +0100
--- repository.c	2017-03-02 20:16:01.352305575 +0100
***************
*** 79,84 ****
--- 79,89 ----
  };
  size_t git_repository__reserved_names_posix_len = 1;
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static void set_odb(git_repository *repo, git_odb *odb)
  {
  	if (odb) {
***************
*** 133,138 ****
--- 138,147 ----
  	}
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  void git_repository__cleanup(git_repository *repo)
  {
  	assert(repo);
***************
*** 2796,2801 ****
--- 2805,2815 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  int git_repository_set_ident(git_repository *repo, const char *name, const char *email)
  {
  	char *tmp_name = NULL, *tmp_email = NULL;
***************
*** 2846,2848 ****
--- 2860,2866 ----
  	repo->submodule_cache = 0;
  	return 0;
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
