*** revparse.c.orig	2016-11-19 11:44:15.314163215 +0100
--- revparse.c	2016-11-19 11:44:48.722355798 +0100
***************
*** 202,207 ****
--- 202,212 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int retrieve_oid_from_reflog(git_oid *oid, git_reference *ref, size_t identifier)
  {
  	git_reflog *reflog;
***************
*** 252,257 ****
--- 257,266 ----
  	return GIT_ENOTFOUND;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int retrieve_revobject_from_reflog(git_object **out, git_reference **base_ref, git_repository *repo, const char *identifier, size_t position)
  {
  	git_reference *ref;
