*** revparse.c.orig	2017-11-21 19:13:58.083751934 +0100
--- revparse.c	2017-11-21 19:13:58.135751754 +0100
***************
*** 203,208 ****
--- 203,213 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int retrieve_oid_from_reflog(git_oid *oid, git_reference *ref, size_t identifier)
  {
  	git_reflog *reflog;
***************
*** 253,258 ****
--- 258,267 ----
  	return GIT_ENOTFOUND;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int retrieve_revobject_from_reflog(git_object **out, git_reference **base_ref, git_repository *repo, const char *identifier, size_t position)
  {
  	git_reference *ref;
