*** smart_pkt.c.orig	2017-02-27 22:43:50.000268176 +0100
--- smart_pkt.c	2017-02-27 22:43:50.060268950 +0100
***************
*** 508,513 ****
--- 508,518 ----
  	return git_buf_put(buf, pkt_flush_str, strlen(pkt_flush_str));
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int buffer_want_with_caps(const git_remote_head *head, transport_smart_caps *caps, git_buf *buf)
  {
  	git_buf str = GIT_BUF_INIT;
***************
*** 558,563 ****
--- 563,572 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  /*
   * All "want" packets have the same length and format, so what we do
   * is overwrite the OID each time.
