*** smart_protocol.c.orig	2016-11-19 16:24:56.233284904 +0100
--- smart_protocol.c	2016-11-19 16:25:35.767314141 +0100
***************
*** 644,649 ****
--- 644,654 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int gen_pktline(git_buf *buf, git_push *push)
  {
  	push_spec *spec;
***************
*** 685,690 ****
--- 690,699 ----
  	return git_buf_oom(buf) ? -1 : 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int add_push_report_pkt(git_push *push, git_pkt *pkt)
  {
  	push_status *status;
