*** smart_protocol.c.orig	2017-11-21 19:13:58.087751920 +0100
--- smart_protocol.c	2017-11-21 19:13:58.144751723 +0100
***************
*** 650,655 ****
--- 650,660 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int gen_pktline(git_buf *buf, git_push *push)
  {
  	push_spec *spec;
***************
*** 691,696 ****
--- 696,705 ----
  	return git_buf_oom(buf) ? -1 : 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int add_push_report_pkt(git_push *push, git_pkt *pkt)
  {
  	push_status *status;
