*** smart_protocol.c.orig	2017-07-09 11:38:26.964018215 +0200
--- smart_protocol.c	2017-07-09 11:38:27.016017901 +0200
***************
*** 647,652 ****
--- 647,657 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int gen_pktline(git_buf *buf, git_push *push)
  {
  	push_spec *spec;
***************
*** 688,693 ****
--- 693,702 ----
  	return git_buf_oom(buf) ? -1 : 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int add_push_report_pkt(git_push *push, git_pkt *pkt)
  {
  	push_status *status;
