*** stash.c.orig	2017-11-21 19:13:58.083751934 +0100
--- stash.c	2017-11-21 19:13:58.137751747 +0100
***************
*** 574,579 ****
--- 574,584 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int retrieve_stash_commit(
  	git_commit **commit,
  	git_repository *repo,
***************
*** 608,613 ****
--- 613,622 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int retrieve_stash_trees(
  	git_tree **out_stash_tree,
  	git_tree **out_base_tree,
***************
*** 764,769 ****
--- 773,783 ----
  		}							\
  	} while(false);
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static int ensure_clean_index(git_repository *repo, git_index *index)
  {
  	git_tree *head_tree = NULL;
***************
*** 787,792 ****
--- 801,810 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  static int stage_new_file(const git_index_entry **entries, void *data)
  {
  	git_index *index = data;
***************
*** 1011,1016 ****
--- 1029,1039 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  int git_stash_drop(
  	git_repository *repo,
  	size_t index)
***************
*** 1067,1072 ****
--- 1090,1099 ----
  	return error;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_stash_pop(
  	git_repository *repo,
  	size_t index,
