*** tree-cache.c.orig	2016-11-19 21:30:25.819508601 +0100
--- tree-cache.c	2016-11-19 21:34:37.212801391 +0100
***************
*** 248,253 ****
--- 248,258 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static void write_tree(git_buf *out, git_tree_cache *tree)
  {
  	size_t i;
***************
*** 261,266 ****
--- 266,275 ----
  		write_tree(out, tree->children[i]);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_tree_cache_write(git_buf *out, git_tree_cache *tree)
  {
  	write_tree(out, tree);
