*** tree-cache.c.orig	2017-11-21 19:13:58.084751930 +0100
--- tree-cache.c	2017-11-21 19:13:58.139751740 +0100
***************
*** 249,254 ****
--- 249,259 ----
  	return 0;
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
+ #endif
+ 
  static void write_tree(git_buf *out, git_tree_cache *tree)
  {
  	size_t i;
***************
*** 262,267 ****
--- 267,276 ----
  		write_tree(out, tree->children[i]);
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
+ 
  int git_tree_cache_write(git_buf *out, git_tree_cache *tree)
  {
  	write_tree(out, tree);
