*** tree-cache.c.orig	2016-11-05 23:23:19.196028954 +0100
--- tree-cache.c	2016-11-05 23:21:33.000026110 +0100
***************
*** 252,258 ****
--- 252,265 ----
  {
  	size_t i;
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wformat"
  	git_buf_printf(out, "%s%c%"PRIdZ" %"PRIuZ"\n", tree->name, 0, tree->entry_count, tree->children_count);
+ #pragma GCC diagnostic pop
+ #else
+ 	git_buf_printf(out, "%s%c%"PRIdZ" %"PRIuZ"\n", tree->name, 0, tree->entry_count, tree->children_count);
+ #endif
  
  	if (tree->entry_count != -1)
  		git_buf_put(out, (const char *) &tree->oid, GIT_OID_RAWSZ);
