*** tsort.c.orig	2016-11-19 21:30:41.603589074 +0100
--- tsort.c	2016-11-19 21:32:19.272089285 +0100
***************
*** 374,379 ****
--- 374,384 ----
  	}
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static int tsort_r_cmp(const void *a, const void *b, void *payload)
  {
  	return ((git__tsort_cmp)payload)(a, b);
***************
*** 383,385 ****
--- 388,394 ----
  {
  	git__tsort_r(dst, size, tsort_r_cmp, cmp);
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
