*** tsort.c.orig	2017-12-13 21:45:51.013436793 +0100
--- tsort.c	2017-12-13 21:45:51.055436635 +0100
***************
*** 373,378 ****
--- 373,383 ----
  	}
  }
  
+ #ifdef _WIN32
+ #pragma GCC diagnostic push
+ #pragma GCC diagnostic ignored "-Wpedantic"
+ #endif
+ 
  static int tsort_r_cmp(const void *a, const void *b, void *payload)
  {
  	return ((git__tsort_cmp)payload)(a, b);
***************
*** 382,384 ****
--- 387,393 ----
  {
  	git__tsort_r(dst, size, tsort_r_cmp, cmp);
  }
+ 
+ #ifdef _WIN32
+ #pragma GCC diagnostic pop
+ #endif
