*** util.c.orig	2015-10-10 16:23:07.374897598 +0200
--- util.c	2015-10-10 16:27:43.594899915 +0200
***************
*** 18,23 ****
--- 18,29 ----
  # include <Shlwapi.h>
  #endif
  
+ /**
+  * Changed all printf to Rprintf to pass 'R CMD check git2r'
+  * 2014-08-19: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ void Rprintf(const char*, ...);
+ 
  void git_strarray_free(git_strarray *array)
  {
  	size_t i;
***************
*** 334,369 ****
  	for (i = 0; i < line_count; ++i) {
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			printf("%02X ", (unsigned char)*line & 0xFF);
  
! 		printf("| ");
  
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			printf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		printf("\n");
  	}
  
  	if (last_line > 0) {
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			printf("%02X ", (unsigned char)*line & 0xFF);
  
  		for (j = 0; j < (LINE_WIDTH - last_line); ++j)
! 			printf("	");
  
! 		printf("| ");
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			printf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		printf("\n");
  	}
  
! 	printf("\n");
  }
  
  #ifdef GIT_LEGACY_HASH
--- 340,375 ----
  	for (i = 0; i < line_count; ++i) {
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			Rprintf("%02X ", (unsigned char)*line & 0xFF);
  
! 		Rprintf("| ");
  
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			Rprintf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		Rprintf("\n");
  	}
  
  	if (last_line > 0) {
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			Rprintf("%02X ", (unsigned char)*line & 0xFF);
  
  		for (j = 0; j < (LINE_WIDTH - last_line); ++j)
! 			Rprintf("	");
  
! 		Rprintf("| ");
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			Rprintf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		Rprintf("\n");
  	}
  
! 	Rprintf("\n");
  }
  
  #ifdef GIT_LEGACY_HASH
