*** util.c	2014-08-19 21:29:20.585207372 +0200
--- util-pass-R-CMD-check-git2r.c	2014-08-19 21:29:48.153208109 +0200
***************
*** 14,19 ****
--- 14,23 ----
  # include <Shlwapi.h>
  #endif
  
+ /* Changed all printf to Rprintf to pass 'R CMD check git2r'*/
+ /* 2014-08-19: Stefan Widgren <stefan.widgren@gmail.com>*/
+ void Rprintf(const char*, ...);
+ 
  void git_strarray_free(git_strarray *array)
  {
  	size_t i;
***************
*** 315,350 ****
  	for (i = 0; i < line_count; ++i) {
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			printf("%02X ", (unsigned char)*line & 0xFF);
  
! 		printf("| ");
  
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			printf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		printf("\n");
  	}
  
  	if (last_line > 0) {
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			printf("%02X ", (unsigned char)*line & 0xFF);
  
  		for (j = 0; j < (LINE_WIDTH - last_line); ++j)
! 			printf("	");
  
! 		printf("| ");
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			printf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		printf("\n");
  	}
  
! 	printf("\n");
  }
  
  #ifdef GIT_LEGACY_HASH
--- 319,354 ----
  	for (i = 0; i < line_count; ++i) {
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			Rprintf("%02X ", (unsigned char)*line & 0xFF);
  
! 		Rprintf("| ");
  
  		line = buffer + (i * LINE_WIDTH);
  		for (j = 0; j < LINE_WIDTH; ++j, ++line)
! 			Rprintf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		Rprintf("\n");
  	}
  
  	if (last_line > 0) {
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			Rprintf("%02X ", (unsigned char)*line & 0xFF);
  
  		for (j = 0; j < (LINE_WIDTH - last_line); ++j)
! 			Rprintf("	");
  
! 		Rprintf("| ");
  
  		line = buffer + (line_count * LINE_WIDTH);
  		for (j = 0; j < last_line; ++j, ++line)
! 			Rprintf("%c", (*line >= 32 && *line <= 126) ? *line : '.');
  
! 		Rprintf("\n");
  	}
  
! 	Rprintf("\n");
  }
  
  #ifdef GIT_LEGACY_HASH
