*** winhttp.c	2014-12-23 12:27:19.841515801 +0100
--- winhttp-build-with-mingw-w64.c	2014-12-23 12:27:24.669515928 +0100
***************
*** 5,10 ****
--- 5,28 ----
   * a Linking Exception. For full terms see the included COPYING file.
   */
  
+ /**
+  * Changes to build and fix compiler warnings with mingw-w64:
+  *  - Define __CRT_STRSAFE_IMPL to fix duplicate extern error from strsafe.h
+  *  - Add CLSID_InternetSecurityManager
+  *  - Add IID_IInternetSecurityManager
+  *  - Remove #pragma comment(lib, "winhttp")
+  *  - Remove #pragma comment(lib, "rpcrt4")
+  *  - Remove unused variable 't' in function 'winhttp_stream_write_single'
+  *  - Remove unused variable 't' in function 'winhttp_stream_write_buffered'
+  *  - Remove unused variable 't' in function 'winhttp_stream_write_chunked'
+  *  - Remove defined but unused 'prefix_http'
+  *  - Use format specifier PRIxZ in git_buf_printf in function 'write_chunk'
+  *  - Initialize 'wide_len' to 0 in function 'apply_basic_credential'
+  *  - Change 'wchar_t *types[]' to 'const WCHAR *types[]' in 'winhttp_stream_connect'
+  *
+  * 2014-08-27: Stefan Widgren <stefan.widgren@gmail.com>
+  */
+ 
  #ifdef GIT_WINHTTP
  
  #include "git2.h"
***************
*** 17,34 ****
  #include "repository.h"
  
  #include <wincrypt.h>
- #pragma comment(lib, "crypt32")
  #include <winhttp.h>
- #pragma comment(lib, "winhttp")
  
  #include <strsafe.h>
  
  /* For IInternetSecurityManager zone check */
  #include <objbase.h>
  #include <urlmon.h>
  
! /* For UuidCreate */
! #pragma comment(lib, "rpcrt4")
  
  #define WIDEN2(s) L ## s
  #define WIDEN(s) WIDEN2(s)
--- 35,51 ----
  #include "repository.h"
  
  #include <wincrypt.h>
  #include <winhttp.h>
  
+ #define __CRT_STRSAFE_IMPL
  #include <strsafe.h>
  
  /* For IInternetSecurityManager zone check */
  #include <objbase.h>
  #include <urlmon.h>
  
! const IID CLSID_InternetSecurityManager = {0x7b8a2d94,0x0ac9,0x11d1,{0x89,0x6c,0x00,0xc0,0x4f,0xb6,0xbf,0xc4}};
! const IID IID_IInternetSecurityManager  = {0x79EAC9EE,0xBAF9,0x11CE,{0x8C,0x82,0x00,0xAA,0x00,0x4B,0xA9,0x0B}};
  
  #define WIDEN2(s) L ## s
  #define WIDEN(s) WIDEN2(s)
***************
*** 43,49 ****
  #define WINHTTP_IGNORE_REQUEST_TOTAL_LENGTH 0
  #endif
  
- static const char *prefix_http = "http://";
  static const char *prefix_https = "https://";
  static const char *upload_pack_service = "upload-pack";
  static const char *upload_pack_ls_service_url = "/info/refs?service=git-upload-pack";
--- 60,65 ----
***************
*** 98,104 ****
  	git_cred_userpass_plaintext *c = (git_cred_userpass_plaintext *)cred;
  	git_buf buf = GIT_BUF_INIT, raw = GIT_BUF_INIT;
  	wchar_t *wide = NULL;
! 	int error = -1, wide_len;
  
  	git_buf_printf(&raw, "%s:%s", c->username, c->password);
  
--- 114,120 ----
  	git_cred_userpass_plaintext *c = (git_cred_userpass_plaintext *)cred;
  	git_buf buf = GIT_BUF_INIT, raw = GIT_BUF_INIT;
  	wchar_t *wide = NULL;
! 	int error = -1, wide_len = 0;
  
  	git_buf_printf(&raw, "%s:%s", c->username, c->password);
  
***************
*** 272,278 ****
  	git_buf buf = GIT_BUF_INIT;
  	char *proxy_url = NULL;
  	wchar_t ct[MAX_CONTENT_TYPE_LEN];
! 	wchar_t *types[] = { L"*/*", NULL };
  	BOOL peerdist = FALSE;
  	int error = -1;
  	unsigned long disable_redirects = WINHTTP_DISABLE_REDIRECTS;
--- 288,294 ----
  	git_buf buf = GIT_BUF_INIT;
  	char *proxy_url = NULL;
  	wchar_t ct[MAX_CONTENT_TYPE_LEN];
! 	const WCHAR *types[] = { L"*/*", NULL };
  	BOOL peerdist = FALSE;
  	int error = -1;
  	unsigned long disable_redirects = WINHTTP_DISABLE_REDIRECTS;
***************
*** 491,497 ****
  	git_buf buf = GIT_BUF_INIT;
  
  	/* Chunk header */
! 	git_buf_printf(&buf, "%X\r\n", len);
  
  	if (git_buf_oom(&buf))
  		return -1;
--- 507,513 ----
  	git_buf buf = GIT_BUF_INIT;
  
  	/* Chunk header */
! 	git_buf_printf(&buf, "%" PRIxZ "\r\n", len);
  
  	if (git_buf_oom(&buf))
  		return -1;
***************
*** 963,969 ****
  	size_t len)
  {
  	winhttp_stream *s = (winhttp_stream *)stream;
- 	winhttp_subtransport *t = OWNING_SUBTRANSPORT(s);
  	DWORD bytes_written;
  	int error;
  
--- 979,984 ----
***************
*** 1053,1059 ****
  	size_t len)
  {
  	winhttp_stream *s = (winhttp_stream *)stream;
- 	winhttp_subtransport *t = OWNING_SUBTRANSPORT(s);
  	DWORD bytes_written;
  
  	if (!s->request && winhttp_stream_connect(s) < 0)
--- 1068,1073 ----
***************
*** 1099,1105 ****
  	size_t len)
  {
  	winhttp_stream *s = (winhttp_stream *)stream;
- 	winhttp_subtransport *t = OWNING_SUBTRANSPORT(s);
  	int error;
  
  	if (!s->request && winhttp_stream_connect(s) < 0)
--- 1113,1118 ----
