*** zconf.h	2014-12-15 11:41:45.751693575 +0100
--- zconf-fix-include-path.h	2014-12-15 11:58:37.155720658 +0100
***************
*** 8,14 ****
  #ifndef ZCONF_H
  #define ZCONF_H
  
! #include "../../src/common.h"
  
  /* Jeez, don't complain about non-prototype
   * forms, we didn't write zlib */
--- 8,14 ----
  #ifndef ZCONF_H
  #define ZCONF_H
  
! #include "../libgit2/common.h"
  
  /* Jeez, don't complain about non-prototype
   * forms, we didn't write zlib */
