--- diff_print.c	2014-08-09 14:39:13.648531050 +0200
+++ diff_print-pass-R-CMD-check-git2r.c	2014-08-09 14:38:54.504530538 +0200
@@ -13,6 +13,15 @@
 #include "delta.h"
 #include "git2/sys/diff.h"
 
+/* Change to pass 'R CMD check git2r'
+ *
+ * Remove stdout from git_diff_print_callback__to_file_handle and
+ * return error code (-1) instead of using stdout if FILE pointer is
+ * NULL.
+ *
+ * 2014-08-09: Stefan Widgren <stefan.widgren@gmail.com>
+*/
+
 typedef struct {
 	git_diff *diff;
 	git_diff_format_t format;
@@ -613,7 +622,10 @@
 	const git_diff_line *line,
 	void *payload)
 {
-	FILE *fp = payload ? payload : stdout;
+	FILE *fp = payload;
+
+        if (!fp)
+            return -1;
 
 	GIT_UNUSED(delta); GIT_UNUSED(hunk);
 
